src/test/java/com/google/cloud/spanner/pgadapter/hibernate/ITHibernateTest.java

```diff
@@ -0,0 +1,152 @@
+// limitations under the License.
+
+package com.google.cloud.spanner.pgadapter.hibernate;
+
+import static org.junit.Assert.assertEquals;
+
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.pgadapter.PgAdapterTestEnv;
+import com.google.common.collect.ImmutableList;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Scanner;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ITHibernateTest {
+  private static final Logger LOGGER = Logger.getLogger(ITHibernateTest.class.getName());
+  private static final String HIBERNATE_SAMPLE_DIRECTORY = "samples/java/hibernate";
+  private static final String HIBERNATE_PROPERTIES_FILE =
+      HIBERNATE_SAMPLE_DIRECTORY + "/src/main/resources/hibernate.properties";
+  private static final String HIBERNATE_DB_CHANGELOG_DDL_FILE =
+      HIBERNATE_SAMPLE_DIRECTORY + "/src/main/resources/sample-schema-sql";
+  private static final String HIBERNATE_DEFAULT_URL =
+      "jdbc:postgresql://localhost:5433/test-database";
+  private static final PgAdapterTestEnv testEnv = new PgAdapterTestEnv();
+  private static Database database;
+  private static String originalHibernateProperties;
+
+  @BeforeClass
+  public static void setup()
+      throws ClassNotFoundException, IOException, SQLException, InterruptedException {
+    // Make sure the PG JDBC driver is loaded.
+    Class.forName("org.postgresql.Driver");
+
+    testEnv.setUp();
+    database = testEnv.createDatabase(ImmutableList.of());
+    testEnv.startPGAdapterServer(ImmutableList.of());
+    // Create databasechangelog and databasechangeloglock tables.
+    StringBuilder builder = new StringBuilder();
+    try (Scanner scanner = new Scanner(new FileReader(HIBERNATE_DB_CHANGELOG_DDL_FILE))) {
+      while (scanner.hasNextLine()) {
+        builder.append(scanner.nextLine()).append("\n");
+      }
+    }
+    // Note: We know that all semicolons in this file are outside of literals etc.
+    String[] ddl = builder.toString().split(";");
+    String url =
+        String.format(
+            "jdbc:postgresql://localhost:%d/%s",
+            testEnv.getPGAdapterPort(), database.getId().getDatabase());
+    try (Connection connection = DriverManager.getConnection(url)) {
+      try (Statement statement = connection.createStatement()) {
+        for (String sql : ddl) {
+          LOGGER.info("Executing " + sql);
+          statement.execute(sql);
+        }
+      }
+    }
+
+    // Write liquibase.properties
```

Comment:
Do a search-and-replace liquibase -> Hibernate ;-)

Reply:
Done